engine/table/src/main/java/io/deephaven/engine/table/impl/select/StatelessSelectColumn.java

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl.select;
+
+import io.deephaven.api.ColumnName;
+import io.deephaven.api.expression.Expression;
+import io.deephaven.engine.rowset.TrackingRowSet;
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.engine.table.WritableColumnSource;
+import io.deephaven.engine.table.impl.BaseTable;
+import io.deephaven.engine.table.impl.MatchPair;
+import io.deephaven.engine.table.impl.QueryCompilerRequestProcessor;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link SelectColumn} implementation that wraps another {@link SelectColumn} and makes it report to be
+ * {@link #isStateless() stateless}.
+ */
+class StatelessSelectColumn implements SelectColumn {
+
+    private final SelectColumn inner;
+
+    StatelessSelectColumn(@NotNull final SelectColumn inner) {
+        this.inner = inner;
+    }
+
+    @Override
+    public List<String> initInputs(TrackingRowSet rowSet, Map<String, ? extends ColumnSource<?>> columnsOfInterest) {
+        return inner.initInputs(rowSet, columnsOfInterest);
+    }
+
+    @Override
+    public List<String> initDef(@NotNull Map<String, ColumnDefinition<?>> columnDefinitionMap) {
+        return inner.initDef(columnDefinitionMap);
+    }
+
+    @Override
+    public List<String> initDef(@NotNull Map<String, ColumnDefinition<?>> columnDefinitionMap,
+            @NotNull QueryCompilerRequestProcessor compilationRequestProcessor) {
+        return inner.initDef(columnDefinitionMap, compilationRequestProcessor);
+    }
+
+    @Override
+    public Class<?> getReturnedType() {
+        return inner.getReturnedType();
+    }
+
+    @Override
+    public Class<?> getReturnedComponentType() {
+        return inner.getReturnedComponentType();
+    }
+
+    @Override
+    public List<String> getColumns() {
+        return inner.getColumns();
+    }
+
+    @Override
+    public List<String> getColumnArrays() {
+        return inner.getColumnArrays();
+    }
+
+    @Override
+    public @NotNull ColumnSource<?> getDataView() {
```

Comment:
I prefer `@NotNull` before `public` 
```suggestion
    @Override
    @NotNull
    public ColumnSource<?> getDataView() {
```